crates/core-crnn/benches/bench.rs

```diff
@@ -0,0 +1,24 @@
+use core_crnn::activation_function::ActivationFunction;
+use core_crnn::thinking_layer::ThinkingLayer;
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+
+pub fn criterion_benchmark(c: &mut Criterion) {
+    let mut small_model = ThinkingLayer::new(5, 25, 5, ActivationFunction::Other(|x| x)).unwrap();
+    let mut medium_model = ThinkingLayer::new(5, 250, 5, ActivationFunction::Other(|x| x)).unwrap();
+    let mut large_model = ThinkingLayer::new(5, 2500, 5, ActivationFunction::Other(|x| x)).unwrap();
```

Comment:
kinda scuffed numbers. maybe make it powers of 2. like input 8 output 4 internal 32, 256 and, 2048. But not so important